lib/node_modules/@stdlib/assert/has-same-constructor/lib/main.js

```diff
@@ -0,0 +1,53 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MAIN //
+
+/**
+* Test whether two provided values have the same constructor.
+*
+* @param {*} a - The first value to test.
+* @param {*} b - The second value to test.
+* @returns {boolean} Returns `true` if both values have the same constructor, else `false`.
+*
+* @example
+* var obj1 = new Date();
+* var obj2 = new Date();
+* console.log(hasSameConstructor(obj1, obj2));
+* // returns true, both are instances of Date
```

Comment:
This is not project convention. Please review and match how we write other packages.